<?xml version="1.0"?>
<robot name="hector_base" xmlns:xacro="http://ros.org/wiki/xacro">

  <!-- ============================  A Few Color Macros  ============================ -->
	
	<xacro:macro name="black">
		<material name="black">
			<color rgba="0.1 0.1 0.1 1.0"/>
		</material>
	</xacro:macro>
	
	<xacro:macro name="white">
		<material name="white">
			<color rgba="1.0 1.0 1.0 1.0"/>
		</material>
	</xacro:macro>
	
	<xacro:macro name="grey1">
		<material name="grey1">
			<color rgba="0.75 0.75 0.75 1.0"/>
		</material>
	</xacro:macro>
	
	<xacro:macro name="grey2">
		<material name="grey2">
			<color rgba="0.6 0.6 0.6 1.0"/>
		</material>
	</xacro:macro>
	
	<xacro:macro name="grey3">
		<material name="grey3">
			<color rgba="0.8 0.8 0.8 1.0"/>
		</material>
	</xacro:macro>

	<xacro:macro name="ax12_color">
		<material name="ax12_color">
			<color rgba="0.2 0.2 0.2 1.0"/>
		</material>
	</xacro:macro>
	
	<xacro:macro name="bracket_color">
		<material name="bracket_color">
			<color rgba="0.9 0.9 0.9 1.0"/>
		</material>
	</xacro:macro>

        <xacro:macro name="cyan1">
		<material name="cyan1">
			<color rgba="0 0.9 0.9 1.0"/>
		</material>
	</xacro:macro>

	<xacro:macro name="cyan2">	
		<material name="cyan2">
	     <color rgba="0 0.7 0.7 1.0"/>
		</material>
	</xacro:macro>
	
	<xacro:macro name="cyan3">	
		<material name="cyan3">
	     <color rgba="0 0.5 0.5 1.0"/>
		</material>
	</xacro:macro>

        <xacro:macro name="yellow2">
		<material name="yellow2">
			<color rgba="0.8 0.8 0 1.0"/>
		</material>
	</xacro:macro>
	
	<xacro:macro name="green1">
		<material name="green1">
			<color rgba="0 1 0 1.0"/>
		</material>
	</xacro:macro>
	
	<xacro:macro name="green2">
		<material name="green2">
			<color rgba="0.1 0.8 0 1.0"/>
		</material>
	</xacro:macro>

        <xacro:macro name="blue1">
		<material name="blue1">
			<color rgba="0 0 0.9 1.0"/>
		</material>
	</xacro:macro>

        <xacro:macro name="ax12_color">
		<material name="ax12_color">
			<color rgba="0.1 0.1 0.1 1.0"/>
		</material>
	</xacro:macro>
	
 	<!-- ============================   A Few Shape Macros  ============================ -->
	<xacro:macro name="ax12" params="name xyz rpy">
		<link name="${name}">
			<visual>
				<origin xyz="${xyz}" rpy="${rpy}" />
				<geometry>
					<mesh filename="package://robbie/meshes/ax12_box2.${MESH_EXT}" scale="${M_SCALE} ${M_SCALE} ${M_SCALE}"/>
				</geometry>
				<xacro:ax12_color />
             </visual>
             
             <collision>
				<origin xyz="${xyz}" rpy="${rpy}" />
				<geometry>
					<!--<box size="0.032 0.038 0.05" />-->
					
					<mesh filename="package://robbie/meshes/ax12_box2.${MESH_EXT}" scale="${M_SCALE} ${M_SCALE} ${M_SCALE}"/>
					
				</geometry>
             </collision>
             <xacro:default_inertial mass="0.055" />   
        </link>
        <xacro:default_gazebo name="${name}" material="Gazebo/FlatBlack" />
	</xacro:macro>
	
	<xacro:macro name="F2_bracket" params="name xyz rpy">
		<link name="${name}">
			<visual>
				<origin xyz="${xyz}" rpy="${rpy}" />
				<geometry>
          			<mesh filename="package://robbie/meshes/F2.${MESH_EXT}" scale="${M_SCALE} ${M_SCALE} ${M_SCALE}"/>
				</geometry>
				<xacro:bracket_color />
             </visual>
             <collision>
				<origin xyz="${xyz}" rpy="${rpy}" />
					<geometry>
						
          				<mesh filename="package://robbie/meshes/F2.${MESH_EXT}" scale="${M_SCALE} ${M_SCALE} ${M_SCALE}"/>
          				
					</geometry>
             </collision>
             <xacro:default_inertial mass="0.02" />   
        </link>
        <xacro:default_gazebo name="${name}" material="Gazebo/WhiteEmissive" />
	</xacro:macro>
	
	<xacro:macro name="F3_bracket" params="name xyz rpy">
		<link name="${name}">
			<visual>
				<origin xyz="${xyz}" rpy="${rpy}" />
				<geometry>
          			<mesh filename="package://robbie/meshes/F3.${MESH_EXT}" scale="${M_SCALE} ${M_SCALE} ${M_SCALE}"/>
				</geometry><material name="grey1">
			<color rgba="0.75 0.75 0.75 1.0"/>
		</material>
				<xacro:bracket_color />
             </visual>
             <collision>
				<origin xyz="${xyz}" rpy="${rpy}" />
					<geometry>
						
          				<mesh filename="package://robbie/meshes/F3.${MESH_EXT}" scale="${M_SCALE} ${M_SCALE} ${M_SCALE}"/>
						
					</geometry>
             </collision>
             <xacro:default_inertial mass="0.01" />   
        </link>
        <xacro:default_gazebo name="${name}" material="Gazebo/WhiteEmissive" />
	</xacro:macro>

	<xacro:macro name="bracket_f3" params="name xyz rpy">
		<link name="${name}">
			<visual>
				<origin xyz="${xyz}" rpy="${rpy}" />
				<geometry>
					<box size="0.025 0.04 0.006" />	
				</geometry>
				<xacro:bracket_color />
             </visual>
             <collision>
				<origin xyz="${xyz}" rpy="${rpy}" />
					<geometry>
						<box size="0.025 0.04 0.006" />
					</geometry>
             </collision>
             <xacro:default_inertial mass="0.02" />   
        </link>
        <xacro:default_gazebo name="${name}" material="Gazebo/WhiteEmissive" />
	</xacro:macro>
	
	<xacro:macro name="bracket2" params="name xyz rpy">
		<link name="${name}">
			<visual>
				<origin xyz="${xyz}" rpy="${rpy}" />
				<geometry>
					<box size="0.05 0.115 0.07" />	
				</geometry>
				<xacro:bracket_color />
             </visual>
             <collision>
				<origin xyz="${xyz}" rpy="${rpy}" />
					<geometry>
					<box size="0.05 0.115 0.07" />	
					</geometry>
             </collision>
             <xacro:default_inertial mass="0.01" />   
        </link>
        <xacro:default_gazebo name="${name}" material="Gazebo/WhiteEmissive" />
	</xacro:macro>
	
        <xacro:macro name="bracket3" params="name xyz rpy">
		<link name="${name}">
			<visual>
				<origin xyz="${xyz}" rpy="${rpy}" />
				<geometry>
					<box size="0.05 0.032 0.005" />	
				</geometry>
				<xacro:bracket_color />
             </visual>
             <collision>
				<origin xyz="${xyz}" rpy="${rpy}" />
					<geometry>
					<box size="0.05 0.032 0.005" />	
					</geometry>
             </collision>
             <xacro:default_inertial mass="0.01" />   
        </link>
        <xacro:default_gazebo name="${name}" material="Gazebo/WhiteEmissive" />
	</xacro:macro>	

        <xacro:macro name="h_beam" params="name xyz rpy">
		<link name="${name}">
			<visual>
				<origin xyz="${xyz}" rpy="${rpy}" />
				<geometry>
					<box size="0.025 0.11 0.01" />	
				</geometry>
				<xacro:bracket_color />
             </visual>
             <collision>
				<origin xyz="${xyz}" rpy="${rpy}" />
					<geometry>
					<box size="0.025 0.11 0.01" />	
					</geometry>
             </collision>
             <xacro:default_inertial mass="0.01" />   
        </link>
        <xacro:default_gazebo name="${name}" material="Gazebo/WhiteEmissive" />
	</xacro:macro>	


        <xacro:macro name="my_box" params="name lwh xyz rpy mass material *rviz_color">
		<link name="${name}">
			<visual>
				<origin xyz="${xyz}" rpy="${rpy}" />
				<geometry>
					<box size="${lwh}" />
				</geometry>
			    <xacro:insert_block name="rviz_color" />             
             </visual>
             <collision>
				<origin xyz="${xyz}" rpy="${rpy}" />
					<geometry>
						<box size="${lwh}" />
					</geometry>
             </collision>
               
        </link>
        <xacro:default_gazebo name="${name}" material="${material}" />
	</xacro:macro>

	<xacro:macro name="my_cylinder" params="name l r xyz rpy mass material *rviz_color">
		<link name="${name}">
			<visual>	
				<origin xyz="${xyz}" rpy="${rpy}" />
				<geometry>
					<cylinder length="${l}" radius="${r}" />
				</geometry>
            	<xacro:insert_block name="rviz_color" />
            </visual>
            <collision>
				<origin xyz="${xyz}" rpy="${rpy}" />
					<geometry>
						<cylinder length="${l}" radius="${r}" />
					</geometry>
            </collision>
            <xacro:default_inertial mass="${mass}" />
        </link>
        <xacro:default_gazebo name="${name}" material="${material}" />
	</xacro:macro>

    <xacro:macro name="default_inertial" params="mass">
		<inertial>
			<mass value="${mass}" />
			<inertia ixx="0.001" ixy="0.0" ixz="0.0"
                     iyy="0.001" iyz="0.0"
                     izz="0.001" />                     
			</inertial>
    </xacro:macro>
    
    <xacro:macro name="default_gazebo" params="name material">
 		<gazebo reference="${name}">
			<material>${material}</material>
			<selfCollide>false</selfCollide>
			<turnGravityOff>true</turnGravityOff>
		</gazebo>
    </xacro:macro>


</robot>
